synapse/media/url_previewer.py

```diff
@@ -464,9 +465,7 @@ async def _download_url(self, url: str, output_stream: BinaryIO) -> DownloadResu
                     # Use a custom user agent for the preview because some sites will only return
                     # Open Graph metadata to crawler user agents. Omit the Synapse version
                     # string to avoid leaking information.
-                    b"User-Agent": [
-                        "Synapse (bot; +https://github.com/matrix-org/synapse)"
-                    ],
+                    b"User-Agent": [self.hs.config.server.url_preview_user_agent],
```

Comment:
Similar to what we do for `max_spider_size` in `__init__`, we can do the same for `url_preview_user_agent`

```suggestion
                    b"User-Agent": [self.url_preview_user_agent],
```